Dockerfile.rhel

```diff
@@ -0,0 +1,65 @@
+FROM centos:7 AS build-env
+LABEL maintainer "Sida Chen <sidchen@redhat.com>"
+
+ARG GIT_TAG
+RUN test -n "$GIT_TAG" # Expect Clair GIT TAG to be non-empty.
+
+# Install Golang by retrieving the binary
+ENV GO_VERSION=1.11.5
+ENV GO_OS=linux
+ENV GO_ARC=amd64
+RUN curl https://dl.google.com/go/go${GO_VERSION}.${GO_OS}-${GO_ARC}.tar.gz --output go.tar.gz
+RUN echo ff54aafedff961eb94792487e827515da683d61a5f9482f668008832631e5d25 go.tar.gz > GOCHECKSUM
+RUN sha256sum -c GOCHECKSUM
+RUN tar -C /usr/local -xzf go.tar.gz > /dev/null
+ENV GOPATH=/go
+ENV PATH=$PATH:/usr/local/go/bin:${GOPATH}/bin
+
+# Verify go verion
+RUN go version
+
+# Install dependencies
+RUN yum install -y git bzr rpm xz gcc
+# musl-dev 
+RUN go get -u github.com/cloudflare/cfssl/cmd/cfssl
+RUN go get -u github.com/cloudflare/cfssl/cmd/cfssljson
+RUN go get -u github.com/coreos/jwtproxy/cmd/jwtproxy
+
+# Build Clair with git tag
+RUN go get -u -d github.com/coreos/clair/cmd/clair
+WORKDIR /go/src/github.com/coreos/clair
+RUN git checkout $GIT_TAG
+RUN go install github.com/coreos/clair/cmd/clair
+
+FROM registry.access.redhat.com/rhel7/rhel
```

Comment:
Can we make everything about this point its own dockerfile and make it a base image?

It'll probably make the rhel and centos dockerfiles look like this:

```dockerfile
FROM quay.io/coreos/clair-jwt:build-env AS build-env
FROM registry.access.redhat.com/rhel7/rhel
...
COPY --from=build-env /go/bin/cfssl /usr/local/bin/cfssl
COPY --from=build-env /go/bin/cfssljson /usr/local/bin/cfssljson
COPY --from=build-env /go/bin/clair /usr/local/bin/clair
COPY --from=build-env /go/bin/jwtproxy /usr/local/bin/jwtproxy
```